### ISSUE TITLE
PWA dynamic attributes

### ISSUE THREAD

**Issue by ajgassner** [vaadin/flow#6449]
We use Spring Boot to configure e.g. the app name dynamically in application.properties.

It's quite hacky and tricky to get an @PWA powered application to work with dynamic app names, because the annotation attributes are compile time constants.

Please support a way for dynamic pwa attributes. The service worker should refresh the content when attributes change.

Currently we use following ugly reflection hack to get dynamic names: https://github.com/gerald24/possy/blob/59f10cc719f17f314889319c2c96e7bc9b4dc7d7/service/src/main/kotlin/net/g24/possy/service/PwaAnnotationModifier.java

**Comment by heruan** [comment]
I agree with this, PWA should be configurable programmatically. Another use case is simpler and does not involve Spring: having a main layout in a dependency module used by many applications, each one with different PWA attributes.
